causalml/inference/tree/models.py

```diff
@@ -1254,13 +1269,22 @@ def fit(self, X, treatment, y):
         for i, treatment_group_key in enumerate(treatment_group_keys):
             self.classes_[treatment_group_key] = i
 
-        # Bootstrap
-        for tree_i in range(len(self.uplift_forest)):
-            bt_index = np.random.choice(len(X), len(X))
-            x_train_bt = X[bt_index]
-            y_train_bt = y[bt_index]
-            treatment_train_bt = treatment[bt_index]
-            self.uplift_forest[tree_i].fit(X=x_train_bt, treatment=treatment_train_bt, y=y_train_bt)
+        self.uplift_forest = (
+            Parallel(n_jobs=self.n_jobs)
+            (delayed(self.bootstrap)(X, treatment, y, tree) for tree in self.uplift_forest)
+        )
+
+        all_importances = [tree.feature_importances_ for tree in self.uplift_forest]
+        self.feature_importances_ = np.mean(all_importances, axis=0)
+        self.feature_importances_ /= self.feature_importances_.sum()  # normalize to add to 1
```

Comment:
Do we need a `self.feature_importances_.sum() > 0` check here as shown in your `compute_feature_importances()` reference to avoid dividing by zero (e.g., when root is pure)? This might be an extreme edge case though

Reply:
Good callout- I left it out because it _should_ be the rare extreme case (also, if the root is pure, the user is not using uplift trees correctly)

Reply:
But yeah if we want to prevent an error from raising, we can add the condition

Reply:
agree your point that it's minor and more of an user error, thanks for confirming!